keylime/tenant.py

```diff
@@ -624,15 +624,17 @@ def do_status(self, listing=False, returnresponse=False, bulk=False):
         if not listing:
             agent_uuid = self.agent_uuid
 
-        response = None
         do_cvstatus = RequestsClient(self.verifier_base_url, self.tls_enabled)
-        if listing and (self.verifier_id is not None):
-            verifier_id = self.verifier_id
+        if listing:
+            verifier_id = ""
+            if self.verifier_id is not None:
+                verifier_id = self.verifier_id
             response = do_cvstatus.get(
                 (f'/v{self.api_version}/agents/?verifier={verifier_id}'),
                 cert=self.cert,
                 verify=False
             )
+            response = response.json()
```

Comment:
I'm not sure we can discard the original `response` object for it's JSON payload so soon. It will work in the cases where everything is up and running correctly, but in failure modes we have to be able to check the HTTP response object before we can rely on there even being a JSON payload. We can pull the JSON out after we've checked the return codes, or at least with some try/catches in case the methods like `json()` fail. And we shouldn't override the `response` object since that's the only way we can guarantee to check return codes.

Reply:
I agree, we should deal with `response` object internally as is, and only printout `json` at the end.